core/pta/rtc.c

```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) 2018, Linaro Limited
+ * Copyright (c) 2021, EPAM Systems. All rights reserved.
+ *
+ * Based on plat-synquacer/rtc_pta.c
+ *
+ */
+
+#include <drivers/rtc.h>
+#include <kernel/pseudo_ta.h>
+#include <rtc_pta_client.h>
+#include <tee_api_defines.h>
+
+#define PTA_NAME "rtc.pta"
+
+static TEE_Result rtc_pta_get_time(uint32_t types,
+				   TEE_Param params[TEE_NUM_PARAMS])
+{
+	struct optee_rtc_time *time = NULL;
+
+	if (types != TEE_PARAM_TYPES(TEE_PARAM_TYPE_MEMREF_OUTPUT,
+				     TEE_PARAM_TYPE_NONE,
+				     TEE_PARAM_TYPE_NONE,
+				     TEE_PARAM_TYPE_NONE)) {
+		DMSG("bad parameters types: 0x%" PRIx32, types);
+		return TEE_ERROR_BAD_PARAMETERS;
+	}
+
+	time = params[0].memref.buffer;
+	if (!time || params[0].memref.size < sizeof(*time))
```

Comment:
Do we need to be flexible with the size?

Reply:
Actually no, I think this could be fixed for the moment and be a "==" check